packages/mermaid-extension/src/index.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+/**
+ * @packageDocumentation
+ * @module mermaid-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { IThemeManager } from '@jupyterlab/apputils';
+import {
+  IMermaidManager,
+  IMermaidMarkdown,
+  MERMAID_CLASS,
+  MERMAID_CODE_CLASS,
+  MermaidManager,
+  MermaidMarkdown,
+  RenderedMermaid
+} from '@jupyterlab/mermaid';
+
+/**
+ * A namespace for mermaid text-based diagram commands.
+ */
+export namespace CommandIDs {
+  export const copySource = 'mermaid:copy-source';
+}
+
+/**
+ * A plugin for the core rendering/cachine of mermaid text-based diagrams
+ */
+const core: JupyterFrontEndPlugin<IMermaidManager> = {
+  id: '@jupyterlab/mermaid-extension:core',
+  description: 'Provides the Mermaid manager.',
+  autoStart: true,
+  optional: [IThemeManager],
+  provides: IMermaidManager,
+  activate: (app: JupyterFrontEnd, themes: IThemeManager) => {
```

Comment:
Looks like this could have been the following since `IThemeManager` is declared as optional:

```ts
activate: (app: JupyterFrontEnd, themes: IThemeManager | null) => {
```

Reply:
Opened https://github.com/jupyterlab/jupyterlab/pull/15003